### ISSUE TITLE
Support for CreateInstance not returning object path

### ISSUE THREAD

**Issue by andy-maier** [pywbem/pywbem#1319]
DSP0200 states in the description of the CreateInstance operation Section 5.4.2.6:

"It [the object path] is returned if one or more of the new keys of the instance are dynamically allocated during creation rather than specified in the request.".

This seems to allow that CreateInstance does not return the INSTANCENAME element in its response.

Pywbem today in its implementation of `CreateInstance()` in `WBEMConnection` assumes that the object path is always returned:

```
            result = self._imethodcall(
                method_name,
                namespace,
                NewInstance=instance,
                **extra)

            instancename = result[0][2][0]

            # CreateInstance returns an INSTANCENAME, which does not have
            # namespace or host. We want to return a path with namespace,
            # so we set it to the target namespace.
            instancename.namespace = namespace

            return instancename
```

We have issue #1015 open to clarify whether this sentence in the spec is serious or an error.

**Discussion point:** This issue is to discuss whether or not we add support for dealing with a non-existing returned object path in the pywbem code.

**Comment by KSchopmeyer** [comment]
Text I put into Mantis #2730

Here is the text:

This is actually a issue with DSP0200 but that spec is not in the list.

DSP0200 states in the description of the CreateInstance operation Section 5.4.26:

"It [the object path] is returned if one or more of the new keys of the instance are dynamically allocated during creation rather than specified in the request.".

This seems to indicate that CreateInstance does not return the INSTANCENAME element in its response.

However it does clearly state that the path must not be returned if the no keys are dynamically created.  Thus, it is not clear that the client can really do anything with this "efficiency".

Today several implementations (OpenPegasus server) do not follow this statement and always return the path of the created instance whether the path was dynamically allocate during creation or not.  Further, at least pywbem and OpenPegasus client assumes that an object path is always returned. Further, at least two clients(pywbem and OpenPegasus) may actually have an issue in that the completely ignore the fact that in some cases the server may not return a path.

The question is whether this is an enforceable rule or should we clarify that there may be cases where the the path is not returned.

**Comment by KSchopmeyer** [comment]
Note: I tried to clarify the issue from the spec point of view but I would assume that given the fact that there are servers that do not correctly follow the rule, we should probably create a path and insert it if the server returns no path.

**Comment by KSchopmeyer** [comment]
In the case where there is no object path returned, we will generate an exception but it is a cryptic exception.  We propose that rather than we create an exception that clarifies that that there was no object path returned and worry about any other fixes if we get an issue from users.

**Comment by andy-maier** [comment]
Will be addressed by PR #1341.
